jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -958,12 +1083,26 @@ ecma_op_object_define_own_property (ecma_object_t *obj_p, /**< the object */
                                                        is_throw);
     }
 
-    case ECMA_OBJECT_TYPE_ARGUMENTS:
+    case ECMA_OBJECT_TYPE_PSEUDO_ARRAY:
     {
-      return ecma_op_arguments_object_define_own_property (obj_p,
+      ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) obj_p;
+
+      if (ext_object_p->u.pseudo_array.type == ECMA_PSEUDO_ARRAY_ARGUMENTS)
+      {
+        return ecma_op_arguments_object_define_own_property (obj_p,
+                                                             property_name_p,
+                                                             property_desc_p,
+                                                             is_throw);
+      }
+      else
+      {
+        return ecma_op_general_object_define_own_property (obj_p,
```

Comment:
There should be an typed array buffer specific handler here, which captures and updates or rejects updates.